src/Core/src/Platform/iOS/MauiScrollView.cs

```diff
@@ -1,14 +1,39 @@
 ﻿using System;
 using System.Diagnostics.CodeAnalysis;
+using System.Linq;
 using CoreGraphics;
+using Foundation;
 using UIKit;
 
 namespace Microsoft.Maui.Platform
 {
-	public class MauiScrollView : UIScrollView, IUIViewLifeCycleEvents
+	public class MauiScrollView : UIScrollView, IUIViewLifeCycleEvents, IMauiPlatformView
 	{
-		public MauiScrollView()
+		bool _invalidateParentWhenMovedToWindow;
+		CGSize _lastContentSize;
+
+		WeakReference<IView>? _reference;
+
+		internal IView? View
 		{
+			get => _reference != null && _reference.TryGetTarget(out var v) ? v : null;
+			set => _reference = value == null ? null : new(value);
+		}
+
+		public override CGSize ContentSize
+		{
+			get => base.ContentSize;
+			set
+			{
+				base.ContentSize = value;
+
+				// If the ContentSize changed, we need to invalidate the measure of the ScrollView
+				if (_lastContentSize != value)
```

Comment:
Ok the test that is failing is Issue14557 with CollectionView2 , basically one should not use a CollectionView inside a ScrollView, but what this shows is that with something that will a very large ContentSize can cause infinite loop here.

Reply:
This has been solved in #26763 and this PR has been rebased.